### DISCUSSION TITLE
We need data to be object-arrays, or at least some way to see the existing strings untruncated.

### DISCUSSION THREAD

**Discussion by Kaosarbitrium** [microsoft/winget-cli#2857] [PowerShell]
Howdy. I've got a few programs in the works where I make simple GUIs for users to interact with existing powershell programs/features in a easier to manage way. Right now a big one I'm working on, that people are asking to deploy as part of their windows-install kit, is one that lets an admin view every appxpackage installed on the computer, make a quick list of them, and uninstall them. Great for removing some of the bloatware and such fresh windows installs come with these days (Spotify, Disney+, ads tracking/telemetry, etc.), in a quick and simple manner. Or, at the very least, getting an overview of what all is on a system. Issue is, some of the packages that come up via get-appxpackages are programs that need to be uninstalled via winget. As such, instead of simply telling winget "remove id [xxx]", I'd like to take advantage of the tool, poll what else is installed on the PC, and let the user decide what they want to keep or not. 
Issue is, winget does not have consistent proper formatting for it's strings. Frequently it will either cut off the last parts of the strings (Microsoft Visual C++ 2008 Redistributable - x64 9.0.30729.6… Microsoft.VCRedist.2008.x64                                9.0.30729.6161) to preserve the existing formatting scheme, or it will replace the end of whatever part of the string with some giberish characters when they overflow their area of the formatting, which still messes up the formatting for the entire section. (Microsoft Visual C++ 2008 Redistributable ΓÇª Microsoft.VCRedist.2008.x64                  9.0.30729.6161) 
I am almost positive that I am not the only person running into issues like this and is thusly being prevented from doing almost any programing/automation with winget. I like the tool and recommend it to every remotely IT/Tech-Savy person I know when we talk about programs on Windows, but unless people are able to use this for something beyond manually searching, installing, and removing programs, it's going to be limited in how much it gets adopted. Right now it's great for updating programs and doing manual management, but we need to fix/overhaul the formatting if we want it to be used in any real numbers/capacity.

**Comment by denelon** [comment]
We're still working on the PowerShell implementation to surface objects that are easier to work with.
